ft_client/freqtrade_client/ft_rest_client.py

```diff
@@ -312,20 +312,47 @@ def forcebuy(self, pair, price=None):
         data = {"pair": pair, "price": price}
         return self._post("forcebuy", data=data)
 
-    def forceenter(self, pair, side, price=None):
+    def forceenter(
+        self,
+        pair,
+        side,
+        price=None,
+        order_type=None,
+        stake_amount=None,
+        leverage=None,
+        enter_tag=None,
+    ):
         """Force entering a trade
 
         :param pair: Pair to buy (ETH/BTC)
         :param side: 'long' or 'short'
         :param price: Optional - price to buy
+        :param order_type: Optional - 'limit' or 'market'
+        :param stake_amount: Optional - stake amount (as a float)
+        :param leverage: Optional - leverage (as a float)
+        :param enter_tag: Optional - entry tag (as a string, default: 'force_enter')
```

Comment:
well the reason this wasn't added so far is that for the CLI/script usage - this will result in 
`script.py ETH/USDT long 3600 limit 20 2 enter_tag`

Now mixing one or two of these is VERY easy - and if you need the last argument - then NONE of the earlier parameters is optional - forcing you to provide a price for the trade, even though you just wanted to set leverage (or enter_tag).

There's no good solution to this i can think of at the moment.

Now i do understand your reason for needing this - which is different to the script and very well justified - but i think we'll either need a reduced solution default for the "plain" script - or a way to pass named parameters (e.g. `script.py ETH/USDT long enter_tag="<sometag>"`).

---

will need to think about how to tackle this a bit - as this is more than just "adding it to this class" - as it's directly exposing all parameters as positional arguments to a script.

now i'm obviously open for ideas / suggestions ;)

Reply:
```python
import argparse
import inspect
import json
import logging
import re
import sys
from pathlib import Path
from typing import Any, Dict

import rapidjson

from freqtrade_client import __version__
from freqtrade_client.ft_rest_client import FtRestClient


logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
)
logger = logging.getLogger("ft_rest_client")


def add_arguments(args: Any = None):
    parser = argparse.ArgumentParser(
        prog="freqtrade-client",
        description="Client for the freqtrade REST API",
    )
    parser.add_argument(
        "command", help="Positional argument defining the command to execute.", nargs="?"
    )
    parser.add_argument("-V", "--version", action="version", version=f"%(prog)s {__version__}")
    parser.add_argument(
        "--show",
        help="Show possible methods with this client",
        dest="show",
        action="store_true",
        default=False,
    )

    parser.add_argument(
        "-c",
        "--config",
        help="Specify configuration file (default: %(default)s). ",
        dest="config",
        type=str,
        metavar="PATH",
        default="config.json",
    )

    # Add named arguments
    parser.add_argument(
        "--pair",
        help="Trading pair (e.g., ETH/USDT)",
        dest="pair",
        type=str,
        default=None,
    )
    parser.add_argument(
        "--side",
        help="Trade side (e.g., long, short)",
        dest="side",
        type=str,
        default=None,
    )
    parser.add_argument(
        "--price",
        help="Price for the trade",
        dest="price",
        type=float,
        default=None,
    )
    parser.add_argument(
        "--ordertype",
        help="Order type (e.g., limit, market)",
        dest="ordertype",
        type=str,
        default=None,
    )
    parser.add_argument(
        "--amount",
        help="Amount to trade",
        dest="amount",
        type=float,
        default=None,
    )
    parser.add_argument(
        "--leverage",
        help="Leverage for the trade",
        dest="leverage",
        type=int,
        default=None,
    )
    parser.add_argument(
        "--enter_tag",
        help="Enter tag for the trade",
        dest="enter_tag",
        type=str,
        default=None,
    )

    pargs = parser.parse_args(args)
    return vars(pargs)


def load_config(configfile):
    file = Path(configfile)
    if file.is_file():
        with file.open("r") as f:
            config = rapidjson.load(
                f, parse_mode=rapidjson.PM_COMMENTS | rapidjson.PM_TRAILING_COMMAS
            )
        return config
    else:
        logger.warning(f"Could not load config file {file}.")
        sys.exit(1)


def print_commands():
    # Print dynamic help for the different commands using the commands doc-strings
    client = FtRestClient(None)
    print("Possible commands:\n")
    for x, _ in inspect.getmembers(client):
        if not x.startswith("_"):
            doc = re.sub(":return:.*", "", getattr(client, x).__doc__, flags=re.MULTILINE).rstrip()
            print(f"{x}\n\t{doc}\n")


def main_exec(args: Dict[str, Any]):
    if args.get("show"):
        print_commands()
        sys.exit()

    config = load_config(args["config"])
    url = config.get("api_server", {}).get("listen_ip_address", "127.0.0.1")
    port = config.get("api_server", {}).get("listen_port", "8080")
    username = config.get("api_server", {}).get("username")
    password = config.get("api_server", {}).get("password")

    server_url = f"http://{url}:{port}"
    client = FtRestClient(server_url, username, password)

    m = [x for x, y in inspect.getmembers(client) if not x.startswith("_")]
    command = args["command"]
    if command not in m:
        logger.error(f"Command {command} not defined")
        print_commands()
        return

    print(json.dumps(getattr(client, command)(*args["command_arguments"])))


def main():
    """
    Main entry point for the client
    """
    args = add_arguments()
    main_exec(args)
`

How about that?

Reply:
well surely not.
that'd mean we must explicitly specify all parameters.

Reply:
@xmatthias yeah this was my thought too after I submitted the PR. a mapping of optional arguments to be parsed sounds like a fair compromise

Reply:
@simwai please stop spamming random PR's with clearly AI generated solutions. It's neither helping - and will only pollute this PR with unrelated comments.

If i want an untested AI solution - I'll ask an AI myself. 
Thanks for your understanding.

Reply:
The PR was not random, because I was the one who triggered froggleston to implement it. I understand that you don't like my non-working AI codes, and will not post them anymore. The code looked legit, but it was obviously untested and not a full working solution, but I thought maybe it could give some inspiration. Seems like I was complete wrong with that assumption...